test/setup/server_setup.go

```diff
@@ -1117,12 +1118,25 @@ func (s *Server) shutdown() error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	if err := s.cmd.Process.Kill(); err != nil {
-		return errors.Wrap(err, "error while shutting down "+s.serverID)
+	if err := s.cmd.Process.Signal(syscall.SIGTERM); err != nil {
+		return errors.Wrapf(err, "error while gracefully shutting down %s", s.serverID)
 	}
 
-	if _, err := s.cmd.Process.Wait(); err != nil {
-		return errors.Wrap(err, "error while shutting down "+s.serverID)
+	done := make(chan error)
+	go func() { done <- s.cmd.Wait() }()
+
+	select {
+	case err := <-done:
+		if err == nil {
+			break
+		}
+		if _, ok := err.(*exec.ExitError); !ok {
+			return errors.Wrapf(err, "error while shutting down %s", s.serverID)
+		}
+	case <-time.After(10 * time.Second):
+		if err := s.cmd.Process.Kill(); err != nil {
+			return errors.Wrapf(err, "error while forcing shutting down %s", s.serverID)
+		}
```

Comment:
This is unrelated to the destroy issue, and belongs to another package. lets do this in another PR?
I think it is ok to Kill, because we do not yet support graceful shutdown. IE we do not catch the signals and do different things for different signals. I am also not sure it is causing any problems to just Wait().